app/modules/asset_groups/metadata.py

```diff
@@ -0,0 +1,323 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=bad-continuation
+"""
+Asset_groups metadata
+Classes for parsing and validating the received metadata in requests
+--------------------------
+"""
+import enum
+import logging
+import os
+
+from flask import current_app
+from flask_login import current_user  # NOQA
+
+log = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+class AssetGroupMetadataError(Exception):
+    def __init__(self, log_message, message=None, status_code=400):
+        self.message = message
+        self.status_code = status_code
+        if self.message is None:
+            self.message = log_message
+        log.warning(f'Failed: {log_message} {self.status_code}')
+
+
+class BaseAssetGroupMetadata(object):
+    def __init__(self, request_json):
+        self.request_json = request_json
+        self.request = {}
+
+    def process_json_req(self):
+        try:
+            self.request.update(self.request_json)
+        except Exception:
+            raise AssetGroupMetadataError('Failed to parse request')
+
+    # Helper for validating the required fields in any level dictionary
+    def _validate_fields(self, dictionary, fields, error_str):
+        for field, field_type, mandatory in fields:
+            if mandatory:
+                if field not in dictionary or not isinstance(
+                    dictionary[field], field_type
+                ):
+                    raise AssetGroupMetadataError(
+                        f'{field} field missing from {error_str}'
+                    )
+                elif (
+                    field_type == list
+                ):  # All mandatory lists must have at least one entry
+                    if len(dictionary[field]) < 1:
+                        raise AssetGroupMetadataError(
+                            f'{field} in {error_str} must have at least one entry'
+                        )
+
+            elif field in dictionary:
+                if not isinstance(dictionary[field], field_type):
+                    raise AssetGroupMetadataError(
+                        f'{field} incorrect type in {error_str}'
+                    )
+
+
+# Class used to process and validate the json data. This json may be received from the frontend or
+# read from a file in the case of a restart. This class creates no DB objects, it just validates
+# what is read and raises an AssetGroupMetadataError on failure
+class CreateAssetGroupMetadata(BaseAssetGroupMetadata):
```

Comment:
This is renamed from what was the AssetGroupMetadata that was in models.py, the idea being that the common stuff now goes in the base class and the Create and the Patch are two derived classes